sympy/solvers/decompogen.py

```diff
@@ -30,6 +31,8 @@ def decompogen(f, symbol):
     [sin(x), sqrt(x), cos(x), x**2 + 1]
     >>> decompogen(x**4 + 2*x**3 - x - 1, x)
     [x**2 - x - 1, x**2 + x]
+    >>> decompogen(Max(sin(x), x**2), x)
+    [Max, sin(x), x**2]
```

Comment:
I don't think that returning the type is a correct output, as you can see in the examples that it should look like `sin(x)`, not `sin`.
At least, it should satisfy the identity like `compogen(decompogen(f(x), x), x) == f(x)`, and I can see that it does not work.

I don't think that it should be able to compose or decompose functions with arguments like `sin(x), x**2`. 
But for cases like `Max(x**2, 3)` decompogen can be possible and I'd expect it to return `decompogen(Max(x**2, 3)) => [Max(x, 3), x**2]` such that 
`compogen([Max(x, 3), x**2], x) => Max(x**2, 3)`

Reply:
I did what @smichr suggested [here](https://github.com/sympy/sympy/pull/13623#discussion_r163208726).  
> I'd expect it to return `decompogen(Max(x**2, 3), x) => [Max(x, 3), x**2]`

How to do that? Because the arguments can't be compared using `Min` or `Max`..

Reply:
I don't think that @smichr's suggestion is a right patch.
You'd have to inspect the `f.args` and find if it only has one expression which `has(symbol)`. If it contains no args which has the specific symbol, or if it has multiple args which has the specific symbol, it should raise an error.